MediaBrowser.Providers/TV/SeriesMetadataService.cs

```diff
@@ -67,6 +67,21 @@ protected override void MergeData(MetadataResult<Series> source, MetadataResult<
 
             var sourceItem = source.Item;
             var targetItem = target.Item;
+            var sourceSeasonNames = sourceItem.SeasonNames;
+            var targetSeasonNames = targetItem.SeasonNames;
+
+            if (replaceData || targetSeasonNames.Count == 0)
+            {
+                targetItem.SeasonNames = sourceSeasonNames;
+            }
+            else if (targetSeasonNames.Count != sourceSeasonNames.Count || !sourceSeasonNames.Keys.All(targetSeasonNames.ContainsKey))
```

Comment:
Is the is second part really necessary?

Reply:
Yes in case the season names got changed. the main "problem" is that the season names are not persisted within the `Series` object.
I tried modifying the getter to either use the local prop if it has any length and if it has none get the season names from the database but it did never work as I wanted.

Might be worth taking another look though because it is a cleaner solution.